centaur/src/main/scala/centaur/json/JsonUtils.scala

```diff
@@ -26,25 +26,43 @@ object JsonUtils {
       * The second handles scatters, where each scatter has its own element in the array. In this case all of the
       * scatters are flattened themselves with the scatter index being inserted into the flattened key name.
       */
-    def flatten(prefix: String = ""): JsObject = jsValue.asJsObject.fields.foldLeft(JsObject.empty) {
-      case (acc, (k, v: JsArray)) if v.isSingleCallArray => acc ++ JsObject(k -> v.elements.head).flatten(prefix)
-      case (acc, (k, v: JsArray)) if v.hasShardIndex =>
-        // The .get on the shardIndex is safe as we know all elements of the array have a shardIndex field
-        acc ++ v.elements.map(_.asJsObject).fold(JsObject.empty) { (x, y) => x ++ y.flatten(s"$k.${y.shardIndex.get}") }
-      case (acc, (k, v: JsArray)) =>
-        v.elements.zipWithIndex.foldLeft(acc) { case (accumulator, (element, idx)) =>
-          val maybePrefix = if (prefix.isEmpty) "" else s"$prefix."
-          element match {
-            case _: JsObject => accumulator.mergeWith(element.flatten(s"$maybePrefix$k.$idx"))
-            case x: JsValue => accumulator + (s"$maybePrefix$k.$idx" -> x)
+    def flatten(prefix: String = ""): JsObject = {
+
+      def flattenShardAndAttempt(k: String, v: JsArray, f: JsObject => String): JsObject = {
+        v.elements.map(_.asJsObject).fold(JsObject.empty) { (x, y) => x ++ y.flatten(s"$k.${f(y)}") }
+      }
+
+      jsValue.asJsObject.fields.foldLeft(JsObject.empty) {
+        case (acc, (k, v: JsArray)) if v.isSingleCallArray => acc ++ JsObject(k -> v.elements.head).flatten(prefix)
+        case (acc, (k, v: JsArray)) if v.hasShardIndex && v.hasAttemptNumber =>
+          /* The .get on the shardIndex and attemptNumber is safe as we know all elements of the array have a
+             shardIndex and attempt field. This conversion will add the attempt number in the flattened key structure
+             to avoid lossy conversion for multiple attempts of the same shard
+          */
+          acc ++
+            flattenShardAndAttempt(k, v, (y: JsObject) => y.shardIndex.get) ++
+            flattenShardAndAttempt(k, v, (y: JsObject) => s"${y.shardIndex.get}.${y.attemptNumber.get}")
+        case (acc, (k, v: JsArray)) if v.hasShardIndex =>
```

Comment:
I would be tempted to try removing this case. I don't think Cromwell will ever produce calls with `shardIndex` but no `attempt`

Reply:
Hmm, that's true. The only reason I kept it is because of the fear of breaking other tests. I will remove it to see what happens